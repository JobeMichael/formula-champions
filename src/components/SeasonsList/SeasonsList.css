.table thead {
  background: #00a7bb;
  color: #fff;
}
.table td,
.table th {
  padding: 0.25rem;
  vertical-align: middle;
  border-top: 1px solid #dee2e6;
}
.table .button-holder {
  text-align: center;
}

.table .button-holder a button {
  background-color: #2c8280;
  border: none;
}
