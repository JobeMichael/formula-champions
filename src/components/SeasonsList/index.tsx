import React from "react";
import { Button, Table } from "react-bootstrap";
import { SeasonList } from "../../services/getSeasons";
import Link from "../UI/Link";

interface SeasonListProps {
  data: Array<SeasonList>;
}

const SeasonsList = ({ data }: SeasonListProps) => {
  return (
    <Table responsive striped bordered hover>
      <thead>
        <tr>
          <th>Season</th>
          <th>Name</th>
          <th>Nationality</th>
          <th>Team</th>
          <th>Wins</th>
          <th>Points</th>
          <th></th>
        </tr>
      </thead>
      <tbody>
        {data.map(
          ({ name, nationality, wins, points, team, season }: SeasonList) => (
            <tr key={season}>
              <td>{season}</td>
              <td>{name}</td>
              <td>{nationality}</td>
              <td>{team}</td>
              <td>{wins}</td>
              <td>{points}</td>
              <td>
                <Link to={`season/${season}`}>
                  <Button>Details</Button>
                </Link>
              </td>
            </tr>
          )
        )}
      </tbody>
    </Table>
  );
};
export default SeasonsList;
