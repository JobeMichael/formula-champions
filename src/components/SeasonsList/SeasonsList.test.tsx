import { render } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router-dom";
import SeasonsList from "./";

const mockData = {
  season: "2006",
  points: "134",
  wins: "7",
  name: "Alonso Fernando",
  nationality: "Spanish",
  team: "Renault",
};

describe("<SeasonsList/>", () => {
  it("renders SeasonsList correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <SeasonsList data={[mockData]} />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
