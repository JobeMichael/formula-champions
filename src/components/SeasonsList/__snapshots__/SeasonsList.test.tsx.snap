// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SeasonsList/> renders SeasonsList correctly 1`] = `
<DocumentFragment>
  <div
    class="table-responsive"
  >
    <table
      class="table"
    >
      <thead>
        <tr>
          <th>
            #
          </th>
          <th>
            Table heading
          </th>
          <th>
            Table heading
          </th>
          <th>
            Table heading
          </th>
          <th>
            Table heading
          </th>
          <th>
            Table heading
          </th>
          <th>
            Table heading
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            2006
          </td>
          <td>
            Alonso Fernando
          </td>
          <td>
            Spanish
          </td>
          <td>
            Renault
          </td>
          <td>
            7
          </td>
          <td>
            134
          </td>
          <td>
            <a
              href="/season/2006"
            >
              <button
                class="btn btn-primary"
                type="button"
              >
                Details
              </button>
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
